airbyte-webapp/src/components/connection/CatalogTree/next/useRowStatus.tsx

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable css-modules/no-unused-class */
+import { faMinus, faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import { useField } from "formik";
+import { isEqual } from "lodash";
+import { useMemo } from "react";
+
+import { ModificationIcon } from "components/icons/ModificationIcon";
+import { PillButtonVariant } from "components/ui/PillSelect/PillButton";
+
+import { SyncSchemaStream } from "core/domain/catalog";
+import { useBulkEditSelect } from "hooks/services/BulkEdit/BulkEditService";
+import { useConnectionFormService } from "hooks/services/ConnectionForm/ConnectionFormService";
+
+import styles from "./CatalogTreeTableRow.module.scss";
+
+export const useRowStatus = (stream: SyncSchemaStream) => {
```

Comment:
The name should be clear enough so that anyone searching for it understand it's specific to the catalogTreeTable. Also this is closer to returning props over statuses, so should use another name.

Reply:
Went with `useCatalogTreeTableRowProps`... a bit verbose but definitely more clear.